pallets/pallet-did-lookup/src/migrations.rs

```diff
@@ -16,28 +16,124 @@
 
 // If you feel like getting in touch with us, you can do so at info@botlabs.org
 
-use crate::{linkable_account::LinkableAccountId, AccountIdOf, Config, ConnectionRecordOf, DidIdentifierOf, Pallet};
-
-use crate::{ConnectedAccounts as ConnectedAccountsV2, ConnectedDids as ConnectedDidsV2};
+use crate::{
+	linkable_account::LinkableAccountId, AccountIdOf, Config, ConnectedAccounts as ConnectedAccountsV2,
+	ConnectedDids as ConnectedDidsV2, ConnectionRecordOf, DidIdentifierOf, MigrationOngoing, Pallet,
+};
 
 use frame_support::{
 	storage_alias,
 	traits::{Get, GetStorageVersion, OnRuntimeUpgrade},
 	Blake2_128Concat,
 };
-use sp_std::{marker::PhantomData, vec};
-
-#[cfg(feature = "try-runtime")]
-use frame_support::traits::OnRuntimeUpgradeHelpersExt;
-#[cfg(feature = "try-runtime")]
-use sp_runtime::SaturatedConversion;
+use sp_std::{marker::PhantomData, vec::Vec};
 
+/// Keytype changed from `AccountId` to `LinkableAccountId` changed in V3
 #[storage_alias]
 type ConnectedDids<T: Config> = StorageMap<Pallet<T>, Blake2_128Concat, AccountIdOf<T>, ConnectionRecordOf<T>>;
+/// Second keytype changed from `AccountId` to `LinkableAccountId` changed in V3
 #[storage_alias]
 type ConnectedAccounts<T: Config> =
 	StorageDoubleMap<Pallet<T>, Blake2_128Concat, DidIdentifierOf<T>, Blake2_128Concat, AccountIdOf<T>, ()>;
 
+/// Migrate the `ConnectedDids` and `ConnectedAccounts` key types for a given
+/// `AccountId`.
+pub(crate) fn do_migrate_account_id<T: Config>(
+	account_id: AccountIdOf<T>,
+	linkable_account: LinkableAccountId,
+) -> Option<DidIdentifierOf<T>> {
+	ConnectedDids::<T>::take(&account_id).map(|did_record| {
+		ConnectedDidsV2::<T>::insert(&linkable_account, did_record.clone());
+		if let Some(v) = ConnectedAccounts::<T>::take(&did_record.did, &account_id) {
+			ConnectedAccountsV2::<T>::insert(&did_record.did, linkable_account, v);
+		}
+		did_record.did
+	})
+}
+
+/// Sanity check that both new typed storage maps `ConnectedDids`,
+/// `ConnectedAccounts` have as many keys as their old counter parts with
+/// different key types.
+pub(crate) fn check_storage_size<T: Config>() -> bool {
```

Comment:
Because of the fact that the two maps are actually the same, I think this check is entirely unnecessary. It will always be true, as far as I can tell. Or am I missing something?

Reply:
Apparently it can be the case, given the test cases. So I resolved this comment.